Examples/Framework/src/Framework/Sequencer.cpp

```diff
@@ -542,6 +552,19 @@ int Sequencer::run() {
               context.fpeMonitor = nullptr;
             }
 
+            if (enableEventTiming) {
+              // Collect the event timing data
+              for (std::size_t i = 0; i < names.size(); i++) {
+                EventTimingInfo eventInfo;
+                eventInfo.identifier = names[i];
+                eventInfo.time_s = std::chrono::duration_cast<Seconds>(
+                                       localClocksAlgorithms[i])
+                                       .count();
+                eventInfo.eventId = event;
+                eventTimingData.push_back(eventInfo);
```

Comment:
I don't think this is thread-safe.

Reply:
From [stackoverflow](https://stackoverflow.com/a/9042750), indeed, it is not.
I am not familiar enough with threading to spot this, thanks!

Reply:
You'll probably need to use a `mutex` to synchronize the vector accesses, but that also might impact the actual performance that you're trying to measure.

Reply:
For now, I made the update be done using unique indices, which, from what I understood, is thread safe.